odoo/models.py

```diff
@@ -1947,24 +1953,33 @@ def _read_group_groupby(self, groupby_spec, query):
         """ Return a pair (<SQL expression>, [<field names used in SQL expression>])
         corresponding to the given groupby element.
         """
-        fname, granularity = parse_read_group_spec(groupby_spec)
+        fname, property_name, granularity = parse_read_group_spec(groupby_spec)
         if fname not in self:
             raise ValueError(f"Invalid field {fname!r} on model {self._name!r}")
 
         field = self._fields[fname]
 
-        if granularity and field.type not in ('datetime', 'date'):
+        if property_name:
+            check_property_field_value_name(property_name)
+            if field.type != "properties":
+                raise ValueError(f"Property set on a non properties field: {property_name!r}")
+
+            fullname = f"{fname}.{property_name}"
+        else:
+            fullname = fname
+
+        if granularity and field.type not in ('datetime', 'date', 'properties'):
             raise ValueError(f"Granularity set on a no-datetime field: {groupby_spec!r}")
```

Comment:
Maybe error message could be more precise, like "non datetime field or property" ?